src/duty/cli.py

```diff
@@ -275,7 +278,18 @@ def main(args: list[str] | None = None) -> int:
     collection.load()
 
     if opts.completion:
-        print(Path(__file__).parent.joinpath("completions.bash").read_text())
+        if opts.completion is True:
+            shell = os.path.basename(os.environ.get('SHELL'))
+        else:
+            shell = opts.completion.lower()
+
+        if shell == 'zsh':
+            print(Path(__file__).parent.joinpath("completions.zsh").read_text())
+        elif shell == 'bash':
+            print(Path(__file__).parent.joinpath("completions.bash").read_text())
+        else:
+            raise NotImplementedError(f"Completion is only supported on Bash and Zsh, got '{shell}'.")
```

Comment:
```suggestion
        try:
            print(Path(__file__).parent.joinpath(f"completions.{shell}").read_text())
        except FileNotFoundError as exc:
            msg = f"Completion is only supported on Bash and Zsh, got '{shell}'."
            raise NotImplementedError(msg) from exc
```

so we don't need to touch this code upon adding support for more shells

Reply:
This is much better. 
Regarding `COMPLETION_DIR` as env var, I don't really see the benefit, can you elaborate on some use-cases?

Reply:
Hmm `COMPLETION_DIR` here is just the folder containing our Python modules. It's therefore not configurable, and we can remove it from the message :thinking: Also it's fine to leave the declaration here, it won't be used anywhere else (I think) so no need to have it as a global variable.

Reply:
It's just a loose idea from my intuition, without a strong argument to it.

This level of configurability can be a huge overkill, sure.

But in my perfect world, every software examining a path for presence of a concrete file would operate similarly to the `PATH` environment variable: the file is looked for subsequently in a bunch of directories.

In this particular case, a real use case would be to allow to fallback to user-defined directories to look up their custom-written completions.

`EXTRA_COMPLETION_DIRS=duty-completions/.` in a project with `duty-completions/` folder.

But yeah, I don't think I can see a person in 100 years that would do it, so feel free to not take this into account.

Also, maybe we should extend the error message even further to mention what it looked for:
```suggestion
        COMPLETION_DIR = Path(__file__).parent
        script_name = f"completions.{shell}"
        try:
            print((COMPLETION_DIR / script_name).read_text())
        except FileNotFoundError as exc:
            msg = f"Completion script not found for {shell!r}, looked for {script_name!r} in {COMPLETION_DIR}"
            raise NotImplementedError(msg) from exc
```

Reply:
> In this particular case, a real use case would be to allow to fallback to user-defined directories to look up their custom-written completions, like EXTRA_DUTY_COMPLETION_DIRS=duty-completions/ in a project with duty-completions/ folder.

But... the `--completion` flag is used to *output* the completion script (that the user can then redirect to a file). If the user already has custom completion scripts somewhere, why would they re-output them through duty?

IMO we shouldn't even say "Completion script not found" as this implies a potential error. Instead we should say "Completions for {shell} shell are not available, feature requests and PRs welcome!".

Reply:
Oops, I haven't seen https://github.com/pawamoy/duty/pull/34#discussion_r1942849094 before https://github.com/pawamoy/duty/pull/34#discussion_r1942870707.

> But... the `--completion` flag is used to output the completion script (that the user can then redirect to a file). If the user already has custom completion scripts somewhere, why would they re-output them through duty?

Yeah, that would be pointless.

Reply:
> IMO we shouldn't even say "Completion script not found" as this implies a potential error. Instead we should say "Completions for {shell} shell are not available, feature requests and PRs welcome!".

Definitely, a good CTA is never bad!

Reply:
152040259ba68317798a6a03b270c712e51e885f